### PR TITLE
Feature/82 location detail rest bundle

### LINKED ISSUE #82: Location detail rest bundle
A location detail rest bundle that can store a selected location Id, and then use that Id to load location detail JSON. For now, just have it load the mock JSON from /mockdata/location-detail.json.

The districtsAndBasins bundle may be a good starting point for how to track the selected location id. However, while the basins use `prefetch: true`, the location detail data can't be fetched until a location Id is stored in the state.

### PR DESCRIPTION
**Pull request by brian428**
Fixes #82 

- Added the logic to handle map overlay click, load location detail, and open the detail sidebar. 
- Added some more mock data to the location-detail.json file for use in the location detail sections on information and dam profile
- Renamed the location map bundle and location summary list bundles for clarity, and decoupled the location detail rest bundle from knowledge of the map bundle. 
- Also some other small tweaks.

### CODE REVIEW

**Review by dlin2102 — APPROVED** [review]
Everything looks good to me. The only thing I would note is that if I'm on the homepage and click on a popup and I go to Map page, the drawer pops out immediately bc it was triggered by the other map. I think that should be fine tho because we are getting rid of the home page map anyway because it lacks purpose.

File: src/app-common/tests/SearchBox.test.js [hunk 1]
```diff
@@ -24,18 +24,23 @@ test("component renders without error", () => {
 });
 
 test("value test", () => {
-  const wrapper = initialSetup( { value: "Hello World" } );
+  const wrapper = initialSetup( { value: "Hello World", key: 1 } );
```
Comment (mohitb3) [inline src/app-common/tests/SearchBox.test.js]: When writing tests using enzyme, its best practice is to use `describe` and `it`. Each `it` function should test one unit of the component, and you group all the `it` tests for one component under a `describe`
```
describe('<SearchBox />', () => {
  const renderOptions = { disableLifecycleMethods: true };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchBox />, renderOptions);
  });

  it('should render component', () => {
    expect(wrapper.length).toBe(1);
  });

  it('should pass value and text props to input', () => {
    wrapper.setProps({
      value: 'Hello World',
      text: 'Some random text',
    });
    expect(wrapper.find('input').prop('value')).toBe('Hello World');
    expect(wrapper.find('input').prop('placeholder')).toBe('Some random text');
  });

  it('should handle onChange() passed to props', () => {
    const onChange = jest.fn();
    wrapper.setProps({ onChange })

    // simulate clicking onChange on the input
    wrapper.find('input').prop('onChange')('testudo');

    expect(onChange).toHaveBeenCalledWith('testudo');
  });

  it('should handle onEnterKey() passed to props', () => {
    const onEnterKey = jest.fn();
    wrapper.setProps({ onEnterKey })

    // simulate clicking onKeyDown with 'Enter' key
    let event = { key: 'Enter' };
    wrapper.find('input').prop('onKeyDown')(event);

    // should call onEnter if the key is 'Enter'
    expect(onEnterKey).toHaveBeenCalledTimes(1);
    expect(onEnterKey).toHaveBeenCalledWith(event);

    // simulate clicking onKeyDown with 'Tab' key
    event = { key: 'Tab' };
    wrapper.find('input').prop('onKeyDown')(event);
    expect(onEnterKey).toHaveBeenCalledTimes(1);
  });

  it('should handle if onChange and onEnterKey are not passed as props', () => {
    wrapper.find('input').prop('onChange')('testudo');
    wrapper.find('input').prop('onKeyDown')({ key: 'Enter '});
  });
});
```
It makes the test output a bit more organized when you're running multiple test files

Screen Shot 2020-10-07 at 3 47 45 PM

Comment (brian428) [reply src/app-common/tests/SearchBox.test.js]: Thanks! Updated all the tests to use that approach.

Comment (mohitb3) [reply src/app-common/tests/SearchBox.test.js]: Great! Everything else looks good

**Review by mohitb3 — APPROVED** [review]

**Pull request merged**